modules/client-configuration/pages/contact-methods-pushssh.adoc

```diff
@@ -0,0 +1,101 @@
+After all actions and events are executed, the tunnel is closed.
+
+The server uses SSH to contact the clients at regular intervals, checking in and performing scheduled actions and events.
+
+
+
+[IMPORTANT]
+====
+Re-installing systems using the provisioning model is not currently supported on clients managed with Push via SSH.
+====
+
+For tunneling connections via SSH, an available port number is required for tunneling via HTTPS.
+The port number used by default is `1233`.
+To overwrite it, you can add a custom port numbers greater than 1024 to [path]``/etc/rhn/rhn.conf``:
+
+----
+ssh_push_port_https = high_port
+----
+
+
+For security reasons, you might want to use sudo with SSH, to access the system as an unprivileged user instead of as root.
+
+
+// FIXME: which commands must we actually list for <user>?
+// FIXME: NOTE probably needs to be adjusted
+.Procedure: Configuring Unprivileged SSH Access
+. Ensure you have the latest [path]``spacewalk-taskomatic`` and [path]``spacewalk-certs-tools`` packages installed on the {productname} Server.
+. On each client system, create an appropriate unprivileged user.
+. On each client system, edit the [filename]``sudoers`` file:
++
+----
+sudo visudo
+----
+. Grant [command]``sudo`` access to the user by adding this line at the end of the [filename]``sudoers`` file.
+  Replace [systemitem]``<user>`` with the name of the user that is bootstrapping the client in the {webui}:
++
+----
+<user>  ALL=NOPASSWD: /usr/bin/python, /usr/bin/python2, /usr/bin/python3, /var/tmp/venv-salt-minion/bin/python
+<user> ALL=(ALL) NOPASSWD:/home/<user>/enable.sh
+<user> ALL=(ALL) NOPASSWD:/home/<user>/bootstrap.sh
+----
++
+[NOTE]
+====
+This procedure grants root access without requiring a password, which is required for registering the client.
+When the client is successfully installed it runs with root privileges, so the access is no longer required.
+We recommend that you remove the line from the [path]``sudoers`` file after the client has been successfully installed.
+====
+
+. On each client system, in the [path]``/home/<user>/.bashrc`` file, add these lines:
++
+----
+PATH=$PATH:/usr/sbin
+export PATH
+----
+. On the {productname} Server, in the [path]``/etc/rhn/rhn.conf`` configuration file, add or amend this line to include the unprivileged username:
++
+----
+ssh_push_sudo_user = <user>
+----
+
+
+Because clients are in the DMZ and cannot reach the server, you need to use the [command]``mgr-ssh-push-init`` tool to register them with the {productname} Server.
+
+To use the tool, you need the client hostname or IP address, and the path to a valid bootstrap script on the {productname} Server.
+For more information about bootstrapping, see xref:client-configuration:registration-bootstrap.adoc[].
+
+The bootstrap script needs to have an activation key associated with it that is configured for Push via SSH.
```

Comment:
I also have no idea how bootstrap script can be used with salt-ssh systems from the client side. Is it really possible somehow?

Reply:
I think bootstrap script cannot be used for salt-ssh. Salt SSH is only possible via Web UI or API call.